impeller/display_list/dl_dispatcher.cc

```diff
@@ -1368,22 +1368,40 @@ Canvas& ExperimentalDlDispatcher::GetCanvas() {
 //// Text Frame Dispatcher
 
 TextFrameDispatcher::TextFrameDispatcher(const ContentContext& renderer,
-                                         const Matrix& initial_matrix)
-    : renderer_(renderer), matrix_(initial_matrix) {}
+                                         const Matrix& initial_matrix,
+                                         const Rect cull_rect)
+    : renderer_(renderer), matrix_(initial_matrix) {
+  cull_rect_state_.push_back(cull_rect);
+}
 
 void TextFrameDispatcher::save() {
   stack_.emplace_back(matrix_);
+  cull_rect_state_.push_back(cull_rect_state_.back());
 }
 
 void TextFrameDispatcher::saveLayer(const DlRect& bounds,
                                     const flutter::SaveLayerOptions options,
                                     const flutter::DlImageFilter* backdrop) {
   save();
+
+  // This dispatcher does not track enough state to accurately compute
+  // cull rects with image filters.
```

Comment:
For this, and other reasons, I think saveLayer should be a NOP.

- First, DL has already done culling against things that appeared within the DL stream when it did the recording. So, the bounds you see here don't necessarily add any new information. The DL dispatch method is already culling things against the cull bounds that were provided in the initial dispatch call. DlBuilder already culled things against the bounds of save layers and clips when it did the recording. So, how is this particular enhancement really adding any value - other than some rare interactions where something might intersect the bounds here and also intersect the cull rect, but somehow not intersect the intersection of the two? If we were testing full geometries then strange cases like that might fall through the holes and we might be able to catch them with some more diligence here, but we are just testing bounds of the geometries, making it less likely to catch a corner case.
- You could more easily do this same kind of calculation in the clip*() methods which don't have the fuzzy issues of saveLayer. Why choose saveLayer to enforce its possibly-fuzzily-potentially clipping when we aren't instrumenting the actual clip calls?
- Really the important thing to track is how the filter manipulates the transform, but we aren't doing that. If anything, we should be setting the cull rect state to MaxRect here (like we have done in the non-experimental AIKS canvas) if there is a filter present, but that is not what the code here is doing...

I'd rather see this method just do:

```
  if (filter) {
    cull_rect_state.back() = Maximum;
  }
```

Reply:
Changed this to set to max cull rect if an image filter is encountered.